src/contract/operations.rs

```diff
@@ -406,7 +323,7 @@ impl Operation for Genesis {
     fn full_type(&self) -> OpFullType { OpFullType::Genesis }
 
     #[inline]
-    fn id(&self) -> OpId { OpId(self.commitment_id().into_inner()) }
+    fn id(&self) -> OpId { self.commit_id() }
```

Comment:
Hi @dr-orlovsky, 

Reading again the audit report, I noticed the OpId, TransitionID, ExtensionID and ContractID need be create based in `OpCommitment` struct instead of `Genesis`, `Transition` and `Extension` directly.

Testing the change locally, now the `contract_id` method an unique ID.

```suggestion
    fn id(&self) -> OpId { self.commit().commit_id() }
```